src/lib/db/query.js

```diff
@@ -17,17 +19,21 @@ const sqlBuilder = (obj, joiner) => {
     ).join(joiner || ' ');
 };
 
+const withoutGroupsCheck = (withoutGroups) => {
+  return withoutGroups && (Array.isArray(withoutGroups) ? searchWithoutGroups : searchWithoutGroup);
```

Comment:
Any reason you've not used the same convention as the rest of queries?
i.e.
```js
  'WHERE': [from && fromDate, ..., withoutGroups && (Array.isArray(withoutGroups) && searchWithoutGroups || searchWithoutGroup)]
```

Reply:
Re-used in multiple queries, so I extracted into function unnecessarily - have refactored as suggestion